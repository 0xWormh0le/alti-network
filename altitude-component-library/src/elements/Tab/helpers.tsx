export const getTabIndexOf = (element: HTMLElement) => {
  const tabs = [].slice
    .call(element?.parentNode?.children)
    .filter((e: HTMLElement) => e.getAttribute('aria-roledescription') === 'tab') as HTMLElement[]
  return tabs.indexOf(element)
}

export const getTabPanelIndexOf = (element: HTMLElement) => {
  const tabPanels = [].slice
    .call(element?.parentNode?.children)
    .filter((e: HTMLElement) => e.getAttribute('aria-roledescription') === 'tabPanel') as HTMLElement[]
  return tabPanels.indexOf(element)
}
