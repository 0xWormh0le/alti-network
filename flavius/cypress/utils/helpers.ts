export const capitalize = (input: string) => {
  if (input.length > 0) {
    const output = input.toLowerCase().split('')
    output[0] = output[0].toUpperCase()
    return output.join('')
  }
  return input
}
