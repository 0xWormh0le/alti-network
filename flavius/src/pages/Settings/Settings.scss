@import 'styles/_shared.scss';

$DEFAULT_PADDING: 22px;

.Settings {
  &__main {
    margin: $PAGE_MARGIN;
    max-width: $MAX_CONTENT_WIDTH;
  }

  &__main &__config-section:not(:last-child) {
    margin-bottom: $DEFAULT_PADDING;
  }

  &__config-section {
    border: 1px solid $PALE_GREY_LIGHT;
    border-radius: 3px;
    padding: $DEFAULT_PADDING;
  }

  &__config-tip {
    display: flex;
    max-width: 28em;
    font-size: 1em;
    padding: 1.2em 1.2em 1.2em 1.5em;
    margin-bottom: 1em;
    border-radius: 0.5em;
    border: 1px solid $PALE_GREY;
    box-shadow: 0px 1px 15px rgba(0, 0, 0, 0.05);

    p {
      margin: 0;
      padding-left: 0.5em;
    }

    img {
      width: 0.9rem;
    }
  }

  &__tab {
    color: inherit;
    cursor: pointer;

    &:hover {
      text-decoration: none;
      color: inherit;
    }
  }

  hr {
    margin-top: $LARGE_SPACING;
    margin-bottom: $LARGE_SPACING;
    border-style: solid;
    border-top-style: none;
    color: $PALE_GREY_LIGHT;
  }
}
