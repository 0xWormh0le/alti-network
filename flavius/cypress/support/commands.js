import * as Sentry from '@sentry/browser'
import { RISK_TYPE_IDS } from '../utils/config'
const { MANY_DOWNLOADS_BY_APP_ID } = RISK_TYPE_IDS
import { version } from '../../package.json'

Cypress.on('fail', (error, runnable) => {
  console.info('we failed!!', error, runnable)
  const errorMessage = `Heartbeat Error detected in spec '${runnable.parent.title}': ${error.message}`
  Sentry.init({
    dsn: 'https://4345e62a3abc48049f9965ccc17ed310@sentry.io/1306610',
    environment: Cypress.env('REACT_APP_STAGE'),
    release: `flavius@${version}`,
    integrations: [
      new Sentry.Integrations.Breadcrumbs({
        console: Cypress.env('REACT_APP_STAGE') === 'production',
        message: errorMessage
      })
    ]
  })
  const err = new Error(errorMessage)
  Sentry.withScope((scope) => {
    scope.setExtra('stacktrace', error.stack)
    scope.setExtra('codeframe', error.codeFrame)
    Sentry.captureException(err)
  })
  throw err
})

const checkMinimumValidPerson = (person) => {
  if (person) {
    expect(person.accessCount).to.be.a('number')
    expect(person.primaryEmail?.address).to.be.a('string')
    expect(person.hasOwnProperty('name')).to.be.true
  }
}

const checkMinimumFile = (file) => {
  const {
    createdAt,
    createdBy,
    externalAccessCount,
    externalAccessList,
    fileId,
    fileName,
    internalAccessCount,
    internalAccessList,
    lastModified,
    linkVisibility,
    lastIngested,
    iconUrl
  } = file
  if (createdAt) {
    expect(createdAt).to.be.a('number')
  }
  if (iconUrl) {
    expect(iconUrl).to.be.a('string')
  }
  expect(externalAccessCount).to.be.a('number')
  expect(externalAccessList).to.be.an('Array')
  // todo: uncomment. id and name is currently null
  expect(fileId).to.be.a('string')
  expect(fileName).to.be.a('string')
  expect(internalAccessCount).to.be.a('number')
  expect(internalAccessList).to.be.an('Array')
  if (lastModified) {
    expect(lastModified).to.be.a('number')
  }
  if (lastIngested) {
    expect(lastIngested).to.be.a('number')
  }
  expect(linkVisibility).to.be.a('string')
}

const checkMinimumFiles = (files) => checkMinimumFile(files[randomIndex(files)])

const randomIndex = (array) => Math.floor(Math.random() * array.length)

Cypress.Commands.add('login', () => {
  cy.get('#email').type('bobbie@thoughtlabs.io')
  cy.get('#password').type(Cypress.env('CYPRESS_USER_PASSWORD'))
  cy.get('.Login__button').click()
})

Cypress.Commands.add('checkMinimumValidPerson', checkMinimumValidPerson)

Cypress.Commands.add('checkFullPerson', (person) => {
  checkMinimumValidPerson(person)
  expect(person.externalCount).to.be.a('number')
  expect(person.userKind).to.be.a('number')
  expect(person.internalCount).to.be.a('number')
})

Cypress.Commands.add('checkFiles', (files) => {
  expect(files).to.be.an('Array')
  checkMinimumFiles(files)
})

Cypress.Commands.add('checkRisks', (risks, filteredRiskTypeId) => {
  expect(risks).to.be.an('Array')
  const randomRiskIndex = randomIndex(risks)
  const randomTestRisk = risks[randomRiskIndex]
  expect(randomTestRisk.platformId).to.be.a('string')
  expect(randomTestRisk.datetime).to.be.a('number')
  expect(randomTestRisk.fileCount).to.be.a('number')
  expect(randomTestRisk.hasOwnProperty('fileId')).to.be.true
  expect(randomTestRisk.hasOwnProperty('fileName')).to.be.true
  expect(randomTestRisk.riskDescription).to.be.a('string')
  expect(randomTestRisk.riskId).to.be.a('string')
  expect(randomTestRisk.riskTarget).to.be.an('Array')
  if (filteredRiskTypeId) {
    expect(randomTestRisk.riskTypeId).to.equal(filteredRiskTypeId)
  }
  expect(randomTestRisk.severity).to.be.a('number')

  checkMinimumValidPerson(randomTestRisk.creator)
  checkMinimumValidPerson(randomTestRisk.owner)

  if (filteredRiskTypeId === MANY_DOWNLOADS_BY_APP_ID) {
    expect(randomTestRisk.plugin.id).to.be.a('string')
    // todo: uncomment. name is currently null
    expect(randomTestRisk.plugin.name).to.be.a('string')
  }
})

Cypress.Commands.add('checkPagination', (paginationResponse, overrides, extra) => {
  expect(paginationResponse.orderBy).to.equal(
    overrides && overrides.hasOwnProperty('orderBy') ? overrides.orderBy : 'datetime'
  )
  expect(paginationResponse.pageCount).to.be.a('number')
  expect(paginationResponse.pageNumber).to.be.a('number')
  expect(paginationResponse.pageSize).to.be.a('number')
  expect(paginationResponse.sort).to.equal(overrides && overrides.hasOwnProperty('sort') ? overrides.sort : 'DESC')

  if (extra) {
    const { wrapper } = extra
    const { pageCount } = paginationResponse
    if (pageCount > 1) {
      if (pageCount > 5) {
        cy.get(wrapper + ' .PaginationItem--number').should('have.length', 5)
      } else {
        cy.get(wrapper + ' .PaginationItem--number').should('have.length', pageCount)
      }
    }
  }
})

Cypress.Commands.add('checkEvents', (events, filesIsObject) => {
  expect(events).to.be.an('Array')
  const randomEventIndex = randomIndex(events)
  const randomTestEvent = events[randomEventIndex]
  expect(randomTestEvent.datetime).to.be.a('number')
  expect(randomTestEvent.eventDescription).to.be.a('string')
  expect(randomTestEvent.eventId).to.be.a('string')
  expect(randomTestEvent.eventName).to.be.a('string')
  expect(randomTestEvent.targetPeople).to.be.an('Array')
  if (!filesIsObject) {
    expect(randomTestEvent.files).to.be.an('Array')
    const randomFileIndex = randomIndex(randomTestEvent.files)
    checkMinimumFile(randomTestEvent.files[randomFileIndex])
  } else {
    checkMinimumFiles(randomTestEvent.files)
  }
})
