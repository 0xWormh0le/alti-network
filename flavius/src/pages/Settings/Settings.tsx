import React from 'react'
import Auth from '@aws-amplify/auth'
import { NavLink, withRouter, RouteComponentProps } from 'react-router-dom'
import { Tabs, Tab, TabList, TabPanel } from '@altitudenetworks/component-library'
import PageTitle from 'components/elements/PageTitle'
import TwoFactorAuthentication from 'components/settings/TwoFactorAuthentication'
import ChangePassword from 'components/settings/ChangePassword'
import BasePage from '../BasePage'
import SlackSettings from 'components/settings/SlackSettings'
import SlackNotifications from 'components/settings/SlackNotifications'
import UI_STRINGS from 'util/ui-strings'
import InformationCard from 'components/elements/InformationCard'
import LightBulb from 'icons/LightBulb'
import './Settings.scss'

interface SettingsState {
  user?: SettingsUser
}

class Settings extends BasePage<RouteComponentProps, SettingsState> {
  protected pageName = this.routePathNames.SETTINGS

  public state = {
    user: undefined
  }

  public async componentDidMount() {
    const user = await Auth.currentAuthenticatedUser()

    this.setState({ user })
  }

  public renderPageContent() {
    const { user } = this.state
    const currentPath = this.props.match.path
    const tabLinks = [`${currentPath}/notifications`, `${currentPath}/accounts`]

    return (
      <div className='Settings'>
        <PageTitle title={UI_STRINGS.SETTINGS.TITLE} />
        {user && (
          <Tabs className='Settings__main' defaultActiveTab={tabLinks.indexOf(this.props.location.pathname)}>
            <TabList>
              <Tab as={NavLink} to={tabLinks[0]} className='Settings__tab'>
                {UI_STRINGS.SETTINGS.NOTIFICATIONS}
              </Tab>
              <Tab as={NavLink} to={tabLinks[1]} className='Settings__tab'>
                {UI_STRINGS.SETTINGS.ACCOUNTS}
              </Tab>
            </TabList>

            <TabPanel>
              <div className='Settings__config-section'>
                <SlackSettings />
                <hr />
                <SlackNotifications />
              </div>
            </TabPanel>
            <TabPanel>
              <InformationCard
                className='Settings__config-tip'
                iconLocation='left'
                icon={<LightBulb size={20} color='black' />}>
                <p>{UI_STRINGS.SETTINGS.ACCOUNTS_NOTE}</p>
              </InformationCard>
              <div className='Settings__config-section'>
                <ChangePassword user={user as unknown as SettingsUser} />
              </div>
              <div className='Settings__config-section'>
                <TwoFactorAuthentication user={user as unknown as SettingsUser} />
              </div>
            </TabPanel>
          </Tabs>
        )}
      </div>
    )
  }
}

export default withRouter(Settings)
