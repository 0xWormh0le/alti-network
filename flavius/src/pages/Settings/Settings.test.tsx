import React from 'react'
import { renderWithRouter } from 'test/support/helpers'
import Settings from './Settings'
import Auth from '@aws-amplify/auth'
import { act } from 'react-dom/test-utils'
import { RenderResult } from '@testing-library/react'

jest.mock('@aws-amplify/auth')

let renderResult: RenderResult

describe('Settings', () => {
  beforeEach(async () => {
    const authUser: any = Auth.currentAuthenticatedUser
    authUser.mockResolvedValue(true)

    await act(async () => {
      renderResult = renderWithRouter(<Settings />, { route: '/settings/accounts', mountRoute: '/settings' })
      await new Promise((r) => setTimeout(r, 100))
    })
  })

  it('renders correctly', async () => {
    const { container } = renderResult
    expect(container).toMatchSnapshot()
  })

  it('has password elements', async () => {
    const { getByText } = renderResult
    expect(getByText(/Current Password/i)).toBeInTheDocument()
    expect(getByText(/Confirm New Password/i)).toBeInTheDocument()
  })
})
