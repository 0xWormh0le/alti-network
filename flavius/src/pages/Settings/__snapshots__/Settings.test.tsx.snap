// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings renders correctly 1`] = `
<div>
  <div
    class="Page"
  >
    <div
      class="Settings"
    >
      <h1
        class="Typography Typography--h2 Typography--normal PageTitle"
      >
        Settings
      </h1>
      <div
        class="Tabs Settings__main"
      >
        <div
          class="Tabs__tab-list"
        >
          <a
            aria-roledescription="tab"
            class="Tabs__tab Settings__tab"
            href="/settings/notifications"
          >
            Notifications
          </a>
          <a
            aria-current="page"
            aria-roledescription="tab"
            class="Tabs__tab Settings__tab Tabs__tab--selected active"
            href="/settings/accounts"
          >
            Accounts
          </a>
        </div>
        <div
          aria-roledescription="tabPanel"
          class="hidden"
        />
        <div
          aria-roledescription="tabPanel"
          class="Tabs__tab-panel  "
        >
          <div
            class="InformationCard Settings__config-tip"
          >
            <div
              class="InformationCard__summary"
            >
              <div
                class="InformationCard__icon InformationCard__icon--left"
              >
                <svg
                  fill="none"
                  height="20"
                  style="padding: 0.15em; background-color: transparent; border-radius: 50%;"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    clip-path="url(#clip0)"
                  >
                    <path
                      d="M9.96129 4.39517C6.82137 4.39517 4.23524 6.96614 4.23528 10.121C4.23528 12.2319 5.39109 14.165 7.25153 15.166C7.4554 15.2757 7.58206 15.4915 7.58206 15.7291V17.9436C7.58206 19.0108 8.45032 19.879 9.51754 19.879H10.4046C11.4719 19.879 12.3401 19.0108 12.3401 17.9436V15.7291C12.3401 15.4923 12.4722 15.2736 12.6849 15.1584C14.5366 14.1553 15.6869 12.2251 15.6869 10.121C15.6869 6.95618 13.1253 4.39517 9.96129 4.39517ZM10.4046 18.5887H9.51754C9.16181 18.5887 8.87238 18.2993 8.87238 17.9436V16.371H11.0498V17.9436C11.0498 18.2993 10.7604 18.5887 10.4046 18.5887ZM12.0703 14.0238C11.6397 14.2571 11.3208 14.6366 11.1622 15.0807H8.76141C8.60464 14.6366 8.28911 14.2591 7.8629 14.0298C6.42121 13.2541 5.5256 11.7563 5.5256 10.121C5.5256 7.68263 7.53673 5.66827 9.99516 5.68557C12.4221 5.7038 14.3966 7.69348 14.3966 10.1209C14.3966 11.751 13.5052 13.2465 12.0703 14.0238ZM9.61702 7.52469C9.7531 7.854 9.59645 8.23126 9.26714 8.36731C8.78399 8.56694 8.39895 8.96065 8.21077 9.44743C8.11185 9.70332 7.86766 9.86013 7.60887 9.86013C7.1573 9.86013 6.84367 9.40529 7.00726 8.98219C7.32222 8.16747 7.96629 7.50872 8.77439 7.17481C9.10375 7.03864 9.48093 7.19537 9.61702 7.52469ZM2.82258 10.1196C2.82258 10.4759 2.53371 10.7647 2.17742 10.7647H0.645161C0.288871 10.7647 0 10.4759 0 10.1196C0 9.76327 0.288871 9.4744 0.645161 9.4744H2.17742C2.53371 9.4744 2.82258 9.76327 2.82258 10.1196ZM9.35625 2.29839V0.766133C9.35625 0.409843 9.64512 0.120972 10.0014 0.120972C10.3577 0.120972 10.6466 0.409843 10.6466 0.766133V2.29839C10.6466 2.65468 10.3577 2.94355 10.0014 2.94355C9.64512 2.94355 9.35625 2.65468 9.35625 2.29839ZM2.92992 3.9613C2.67794 3.70936 2.67794 3.30085 2.92992 3.04892C3.18185 2.79694 3.59036 2.79694 3.8423 3.04892L4.92577 4.13239C5.33351 4.54009 5.03903 5.23372 4.4696 5.23372C4.30448 5.23372 4.13939 5.17073 4.01343 5.04472L2.92992 3.9613ZM15.0762 4.13436L16.1597 3.05089C16.4116 2.79892 16.8201 2.79892 17.0721 3.05089C17.324 3.30283 17.324 3.71134 17.0721 3.96327L15.9886 5.04674C15.8626 5.17275 15.6975 5.23573 15.5324 5.23573C14.9628 5.23573 14.6685 4.54198 15.0762 4.13436ZM20 10.1224C20 10.4787 19.7111 10.7676 19.3548 10.7676H17.8226C17.4663 10.7676 17.1774 10.4787 17.1774 10.1224C17.1774 9.7661 17.4663 9.47723 17.8226 9.47723H19.3548C19.7111 9.47723 20 9.76606 20 10.1224Z"
                      fill="#000000"
                    />
                  </g>
                </svg>
              </div>
              <span />
            </div>
            <div
              class="InformationCard__details InformationCard__details--visible"
              style="max-height: 0px;"
            >
              <p>
                The settings on this page is only applicable if you choose to log in with the Altitude Networks’ username and password option.
              </p>
            </div>
          </div>
          <div
            class="Settings__config-section"
          >
            <div
              class="ChangePassword"
            >
              <h3
                class="Typography Typography--h3 SectionTitle"
              >
                Change Password
              </h3>
              <form
                class="ChangePassword__form"
              >
                <div
                  class="FormGroup"
                >
                  <label
                    class="FormLabel"
                    for="oldPassword"
                  >
                    Current Password
                  </label>
                  <input
                    autocomplete="current-password"
                    class="FormControl FormControl--flat"
                    id="oldPassword"
                    placeholder="•••••"
                    type="password"
                    value=""
                  />
                </div>
                <hr
                  class="ChangePassword__separator"
                />
                <div
                  class="FormGroup"
                >
                  <label
                    class="FormLabel"
                    for="password"
                  >
                    New Password
                  </label>
                  <input
                    autocomplete="new-password"
                    class="FormControl FormControl--flat"
                    id="password"
                    placeholder="•••••"
                    type="password"
                    value=""
                  />
                  <div>
                    <svg
                      class="SetPasswordField__toggle-plaintext"
                      style="enable-background: new 0 0 59.2 59.2;"
                      version="1.1"
                      viewBox="0 0 59.2 59.2"
                      x="0px"
                      xmlns="http://www.w3.org/2000/svg"
                      y="0px"
                    >
                      <path
                        d="M51.062 21.841a30.62 30.62 0 0 0-6.99-5.235l9.549-9.549a.999.999 0 1 0-1.414-1.414l-10.021 10.02a30.386 30.386 0 0 0-12.685-2.753c-8.145 0-15.802 3.171-21.561 8.931L0 29.781l8.138 8.138a30.62 30.62 0 0 0 6.714 5.078l-9.145 9.145a.999.999 0 1 0 1.414 1.414l9.593-9.593a30.39 30.39 0 0 0 12.984 2.886c8.145 0 15.802-3.171 21.561-8.931l7.941-7.941-8.138-8.136zM16.321 41.529a28.59 28.59 0 0 1-6.769-5.024l-6.724-6.724 6.527-6.527c5.381-5.381 12.536-8.345 20.146-8.345 3.903 0 7.683.784 11.169 2.27l-24.349 24.35zm33.524-5.024c-5.381 5.381-12.536 8.345-20.146 8.345-4.015 0-7.9-.833-11.468-2.402l24.361-24.361a28.55 28.55 0 0 1 7.056 5.168l6.724 6.724-6.527 6.526z"
                      />
                      <path
                        d="M29.572 16.85c-7.168 0-13 5.832-13 13 0 2.543.745 4.91 2.012 6.916l17.904-17.904a12.907 12.907 0 0 0-6.916-2.012zM22.459 40.718a12.92 12.92 0 0 0 7.113 2.131c7.168 0 13-5.832 13-13a12.92 12.92 0 0 0-2.131-7.113L22.459 40.718z"
                      />
                    </svg>
                  </div>
                  <div
                    class="SetPasswordField__strength-indicator-wrapper"
                  >
                    <div
                      class="SetPasswordField__strength-label SetPasswordField__strength-label--"
                    />
                    <div
                      class="SetPasswordField__strength-indicator SetPasswordField__strength-indicator--"
                    />
                  </div>
                  <ul
                    class="SetPasswordField__helper-text"
                  >
                    <li>
                      Passwords must be 10 or more characters.
                    </li>
                    <li>
                      Passwords cannot be 'weak'
                    </li>
                    <li>
                      Enabling 2FA greatly improves security.
                    </li>
                  </ul>
                </div>
                <div
                  class="FormGroup"
                >
                  <label
                    class="FormLabel"
                    for="confirmPassword"
                  >
                    Confirm New Password
                  </label>
                  <input
                    autocomplete="new-password"
                    class="FormControl FormControl--flat"
                    id="confirmPassword"
                    placeholder="•••••"
                    type="password"
                    value=""
                  />
                </div>
                <button
                  class="Button ChangePassword__button Button--primary Button--disabled"
                  disabled=""
                  type="submit"
                >
                  Save changes
                </button>
              </form>
            </div>
          </div>
          <div
            class="Settings__config-section"
          >
            <div
              class="TwoFactorAuthentication"
            >
              <h3
                class="Typography Typography--h3 SectionTitle"
              >
                Two-Factor Authentication (2FA)
              </h3>
              <p>
                Set up 2FA using Time-based One-time Password (TOTP)
              </p>
              <button
                class="Button Button--primary"
                type="button"
              >
                Disable 2FA for this user
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
